mitmproxy/web/app.py

```diff
@@ -338,6 +340,40 @@ def get(self, flow_id, message):
         self.set_header("X-Frame-Options", "DENY")
         self.write(message.raw_content)
 
+class FlowContentView(RequestHandler):
+
+    def get(self, flow_id, message, content_view):
+        message = getattr(self.flow, message)
+
+        original_cd = message.headers.get("Content-Disposition", None)
+        filename = None
+        if original_cd:
+            filename = re.search("filename=([\w\" \.\-\(\)]+)", original_cd)
+            if filename:
+                filename = filename.group(1)
+        if not filename:
+            filename = self.flow.request.path.split("?")[0].split("/")[-1]
+
+        filename = re.sub(r"[^\w\" \.\-\(\)]", "", filename)
+        cd = "attachment; filename={}".format(filename)
+        self.set_header("Content-Disposition", cd)
+        self.set_header("Content-Type", "application/json")
+        self.set_header("X-Content-Type-Options", "nosniff")
+        self.set_header("X-Frame-Options", "DENY")
+
+        self.set_header("Content-Encoding", "")
```

Comment:
This endpoint is just emitting JSON, so I believe we don't need any of the stuff above. For example, the Content-Disposition is useful for download filenames. The view will not be downloaded, only the original content.